File: Python/_0810_Chalkboard_XOR_Game/Solution.py
# coding=utf-8
# Date: 2021/5/22 9:37
from typing import List


# 执行用时：80 ms, 在所有 Python3 提交中击败了93.94%的用户
# 内存消耗：14.8 MB, 在所有 Python3 提交中击败了96.97%的用户
class Solution:
    def xorGame(self, nums: List[int]) -> bool:
        s = 0
        for x in nums:
            s ^= x
        return s == 0 or len(nums) % 2 == 0


if __name__ == "__main__":
    print(Solution().xorGame([1, 1, 2]))  # False

File: Java/src/_0810_Chalkboard_XOR_Game/Solution.java
package _0810_Chalkboard_XOR_Game;

/**
 * Date: 2021/5/22 9:33
 * 执行用时：0 ms, 在所有 Java 提交中击败了100.00%的用户
 * 内存消耗：37.8 MB, 在所有 Java 提交中击败了93.90%的用户
 */
public class Solution {

    public boolean xorGame(int[] nums) {
        int s = 0;
        for (int x : nums) s ^= x;
        return s == 0 || nums.length % 2 == 0;
    }

    public static void main(String[] args) {

        int[] nums = {1, 1, 2};
        System.out.println((new Solution()).xorGame(nums));  // false
    }
}
